smart_open/tests/test_smart_open.py

```diff
@@ -194,8 +195,9 @@ def test_open_with_keywords_explicit_r(self):
     @mock_s3
     def test_read_never_returns_none(self):
         """read should never return None."""
-        conn = boto.connect_s3()
-        conn.create_bucket("mybucket")
+        s3 = boto3.resource('s3')
+        s3.create_bucket(Bucket='mybucket')
```

Comment:
Maybe, better to use "global" bucket /key (as in https://github.com/RaRe-Technologies/smart_open/blob/master/smart_open/tests/test_s3.py#L16) + same `maybe_mock_s3`?